src/pseudopeople/interface.py

```diff
@@ -95,22 +91,14 @@ def _coerce_dtypes(data: pd.DataFrame, dataset: Dataset):
     return data
 
 
-def _load_data_from_path(data_path: Path, user_filters: List[Tuple]):
+def _load_data_from_path(
+    data_path: Union[Path, dict], user_filters: List[Tuple]
+) -> pd.DataFrame:
     """Load data from a data file given a data_path and a year_filter."""
-    if data_path.suffix == ".parquet":
-        if len(user_filters) == 0:
-            # pyarrow.parquet.read_table doesn't accept an empty list
-            user_filters = None
-        data = pq.read_table(data_path, filters=user_filters).to_pandas()
+    if isinstance(data_path, dict):
+        data = load_and_prep_1040_data(data_path, user_filters)
```

Comment:
nit: This is a 1040-specific method name but the `if` logic is only that date_path is a dict. Is there an easy way to clear that up a bit?

Reply:
This is the main split between the 1040 and the other datasets. Here data_path will either be a path or a dict. load_and_prep_1040_data is going to call load_standard_dataset_file for each filepath in the dict but the return of this function needs to be a dataframe and this is where we need to format the 1040.